src/util/CMakeLists.txt

```diff
@@ -0,0 +1,13 @@
+add_library(util OBJECT)
+
+set(OPENSMT_FULL_VERSION "\"${OPENSMT_VERSION}\"")
+set(GITHUB_FULL_COMMIT "\"${GIT_COMMIT}\"")
+configure_file(opensmt_version.h.cmake.in ${CMAKE_CURRENT_SOURCE_DIR}/opensmt_version.h)
```

Comment:
What's the benefit of doing it this way compared to just defining a macro?

Reply:
I'm taking the time to write here that I don't know :)

I guess I want to say that even if in my suggestions in #807 I'm generating the file like @BritikovKI does in the original suggestion, I don't mind if it's a macro.

Reply:
I was following the route implemented in `Z3`. That's how they are extracting the version into the code...
I also don't mind macro, but what is the benefit there?)